### PR TITLE
conf-mbedtls: add support for Ubuntu family

### PR DESCRIPTION
**Pull request by jmid**

### CODE REVIEW

**Comment by jmid** [comment]
CI fails on Oracle and macOS.
- 28373b6 installs the corresponding homebrew package https://formulae.brew.sh/formula/mbedtls#default
- 4862207 adds an `x-ci-accept-failures` for Oracle

Since the `build` test is a compiler invocation, I'm unsure if the current one will do on macOS... :crossed_fingers:

**Comment by jmid** [comment]
CI is failing on macOS homebrew ARM64 where files are installed in /opt/homebrew instead.
1bfe6c4 adjusts `build` accordingly.

**Comment by mseri** [comment]
Seems to be still failing on macos

File: packages/conf-mbedtls/conf-mbedtls.1/opam [hunk 1]
```diff
@@ -5,18 +5,25 @@ homepage: "https://tls.mbed.org/"
 bug-reports: "https://github.com/ARMmbed/mbedtls/issues"
 license: "Apache-2.0"
 build: [
-  ["cc" "-I/usr/local/include" "test.c"]
+  ["cc" "-I/usr/local/include" "test.c"] {os != "macos" | os-distribution != "homebrew"}
+  ["cc" "-I$(brew --prefix mbedtls)/include" "test.c"] {os = "macos" & os-distribution = "homebrew"}
```
Comment (mseri) [inline packages/conf-mbedtls/conf-mbedtls.1/opam:9]: I think we need to run it with sh:
```suggestion
  ["sh" "cc -I\"$(brew --prefix mbedtls)/include\" test.c"] {os = "macos" & os-distribution = "homebrew"}
```

File: packages/conf-mbedtls/conf-mbedtls.1/opam [hunk 2]
```diff
@@ -5,18 +5,25 @@ homepage: "https://tls.mbed.org/"
 bug-reports: "https://github.com/ARMmbed/mbedtls/issues"
 license: "Apache-2.0"
 build: [
-  ["cc" "-I/usr/local/include" "test.c"]
+  ["cc" "-I/usr/local/include" "test.c"] {os != "macos" | os-distribution != "homebrew"}
+  ["sh" "cc -I\"$(brew --prefix mbedtls)/include\" test.c"] {os = "macos" & os-distribution = "homebrew"}
```
Comment (mseri) [inline packages/conf-mbedtls/conf-mbedtls.1/opam:9]:
```suggestion
  ["sh" "-c" "cc -I\"$(brew --prefix mbedtls)/include\" test.c"] {os = "macos" & os-distribution = "homebrew"}
```

**Comment by jmid** [comment]
All green! Thanks for your help getting this across the line :pray:

**Comment by mseri** [comment]
Thanks to you!

**Pull request merged**